datasets/hate_offensive/hate_offensive.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Automated Hate Speech Detection and the Problem of Offensive Language."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import datasets
+
+
+_CITATION = """
+@article{article,
+author = {Davidson, Thomas and Warmsley, Dana and Macy, Michael and Weber, Ingmar},
+year = {2017},
+month = {03},
+pages = {},
+title = {Automated Hate Speech Detection and the Problem of Offensive Language}
+}
+"""
+
+_DESCRIPTION = "This dataset contains annotated tweets for automated ate-speech recognition"
+
+_HOMEPAGE = "https://arxiv.org/abs/1905.12516"
+
+_LICENSE = "MIT License"
+
+_URLs = "https://github.com/t-davidson/hate-speech-and-offensive-language/raw/master/data/labeled_data.csv"
+
+
+class HateOffensive(datasets.GeneratorBasedBuilder):
+    """Automated Hate Speech Detection and the Problem of Offensive Language """
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "total_annotation_count": datasets.Value("int32"),
+                "hate_speech_annotations": datasets.Value("int32"),
+                "offensive_language_annotations": datasets.Value("int32"),
+                "neither_annotations": datasets.Value("int32"),
+                "class": datasets.ClassLabel(names=["0", "1", "2"]),
```

Comment:
It would be better to have meaningful names here, such as:
```suggestion
                "class": datasets.ClassLabel(names=[" hate speech", "offensive language", "neither"]),
```